database.go

```diff
@@ -0,0 +1,16 @@
+package main
+
+type Database interface {
+	saveDevice(device Device) (Device, error)
```

Comment:
Interfaces in golang should be created when needed and preferably, where needed. So in this case I would put this interface in `device_service.go` because it is needed only there so that you can write tests.

Additionally, public interface with private method is rather strange. I think it can be reversed - the interface is needed only locally for the service therefore it can be private. The method itself can come from different packages potentially so I think it should be public.